src/Sarif/Visitors/RebaseUriVisitor.cs

```diff
@@ -0,0 +1,187 @@
+        {
+            Run newRun = base.VisitRun(node);
+
+            if (newRun.Files != null)
+            {
+                FixFiles(newRun);
+            }
+            
+            if (node.Properties == null)
+            {
+                node.Properties = new Dictionary<string, SerializedPropertyInfo>();
+            }
+
+            // If the dictionary doesn't exist, we should add it to the properties.  If it does, we should add/update the existing dictionary.
+            Dictionary<string, Uri> baseUriDictionary = new Dictionary<string, Uri>();
+            if (node.Properties.ContainsKey(BaseUriDictionaryName))
+            {
+                if (!TryDeserializePropertyDictionary(node.Properties[BaseUriDictionaryName], out baseUriDictionary))
+                {
+                    // If for some reason we don't have a valid dictionary in the originalUriBaseIds, we move it to another location.
+                    node.Properties[BaseUriDictionaryName + IncorrectlyFormattedDictionarySuffix] = node.Properties[BaseUriDictionaryName];
+                    baseUriDictionary = new Dictionary<string, Uri>();
+                }
+            }
+            
+            // Note--this is an add or update, so if this is run twice with the same base variable, we'll replace the path.
+            baseUriDictionary[_baseName] = _baseUri;
+            
+            newRun.Properties[BaseUriDictionaryName] = ReserializePropertyDictionary(baseUriDictionary);
+
+            return newRun;
+        }
+
+        /// <summary>
+        /// If we are changing the URIs in Results to be relative, we need to also change the URI keys in the files dictionary
+        /// to be relative.
+        /// 
+        /// For FileData, we need to fix up the URI data (making it relative to the appropriate base address), 
+        /// and also fix up the ParentKey (as we are patching the keys in the files dictionary in the Run).
+        /// (We need to fix up the keys as we are patching the PhysicalLocation in the Result objects.)
+        /// </summary>
+        /// <param name="run">A run to fix the Files dictionary of.</param>
+        internal void FixFiles(Run run)
+        {
+            Dictionary<string, FileData> newDictionary = new Dictionary<string, FileData>();
+
+            foreach (var key in run.Files.Keys)
+            {
+                Uri oldUri;
+                string newKey = key;
+                FileData data = run.Files[key];
+                // If the old uri is absolute and we need to rebase it, we should.
+                if (Uri.TryCreate(key, UriKind.Absolute, out oldUri) && oldUri.IsAbsoluteUri && _baseUri.IsBaseOf(oldUri))
+                {
+                    Uri newUri = _baseUri.MakeRelativeUri(oldUri);
+
+                    // Ensure the filedata reflects the correct base URI details.
+                    if (data != null)
+                    {
+                        if (data.ParentKey != null)
+                        {
```

Comment:
consider eliminating nesting and improving readability by doing something like

if (data == null || data.ParentKey == null) { continue; }

I actually will put this sort of thing on the same line as above.